File: android/src/org/artcompany/clock/Vibration.java
package org.artcompany.clock;

import android.app.Activity;
import android.os.Vibrator;
import android.os.Bundle;
import android.content.Intent;
import android.content.Context;


public class Vibration{
    static Vibrator vibrator;
    static Vibration vib;

    public Vibration(){
	vib = this;
	}


    public static void Vibrate(Context context,int x){
	vibrator = (Vibrator)context.getSystemService(Context.VIBRATOR_SERVICE);
	vibrator.vibrate(x);
    }

}

File: source/helpers/vibration.cpp
#include "headers/helpers/vibration.h"
#if defined(Q_OS_ANDROID)
#include<QtAndroid>
#include<QtAndroidExtras>
#include<QAndroidJniObject>
#endif

Vibration::Vibration(QObject *parent)
    : QObject{parent}
{}

Vibration::~Vibration()
{}

void Vibration::vibrate(int x)
{
#if defined (Q_OS_ANDROID)
    QAndroidJniObject::callStaticMethod<void>(
                "org/artcompany/clock/Vibration",
                "Vibrate",
                "(Landroid/content/Context;Ljava/lang/String;)V",
                 QtAndroid::androidContext().object(),
                x);
#endif
}
